File: SampleCode/RestApi/Clients/Java/src/main/java/io/swagger/client/model/AttributesValueCondition.java
/*
 * Emby Server REST API
 * 
 */

package io.swagger.client.model;

import java.util.Objects;
import java.util.Arrays;
import io.swagger.v3.oas.annotations.media.Schema;
import com.google.gson.annotations.SerializedName;
import java.io.IOException;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;

/**
 * Conditions for comparing with a target value.  
 */
@JsonAdapter(AttributesValueCondition.Adapter.class)
public enum AttributesValueCondition {
  ISEQUAL("IsEqual"),
  ISNOTEQUAL("IsNotEqual"),
  ISGREATER("IsGreater"),
  ISGREATEROREQUAL("IsGreaterOrEqual"),
  ISLESS("IsLess"),
  ISLESSOREQUAL("IsLessOrEqual");

  private String value;

  AttributesValueCondition(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static AttributesValueCondition fromValue(String input) {
    for (AttributesValueCondition b : AttributesValueCondition.values()) {
      if (b.value.equals(input)) {
        return b;
      }
    }
    return null;
  }

  public static class Adapter extends TypeAdapter<AttributesValueCondition> {
    @Override
    public void write(final JsonWriter jsonWriter, final AttributesValueCondition enumeration) throws IOException {
      jsonWriter.value(String.valueOf(enumeration.getValue()));
    }

    @Override
    public AttributesValueCondition read(final JsonReader jsonReader) throws IOException {
      Object value = jsonReader.nextString();
      return AttributesValueCondition.fromValue((String)(value));
    }
  }
}

File: SampleCode/RestApi/Net.RestSharp/EmbyClient.Dotnet/Model/AttributesValueCondition.cs
/*
 * EmbyClient.Dotnet
 */

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using SwaggerDateConverter = EmbyClient.Dotnet.Client.SwaggerDateConverter;

namespace EmbyClient.Dotnet.Model
{
    /// <summary>
    /// Conditions for comparing with a target value.  
    /// </summary>
    /// <value>Conditions for comparing with a target value.  </value>
    [JsonConverter(typeof(StringEnumConverter))]
        public enum AttributesValueCondition
    {
        /// <summary>
        /// Enum IsEqual for value: IsEqual
        /// </summary>
        [EnumMember(Value = "IsEqual")]
        IsEqual = 1,
        /// <summary>
        /// Enum IsNotEqual for value: IsNotEqual
        /// </summary>
        [EnumMember(Value = "IsNotEqual")]
        IsNotEqual = 2,
        /// <summary>
        /// Enum IsGreater for value: IsGreater
        /// </summary>
        [EnumMember(Value = "IsGreater")]
        IsGreater = 3,
        /// <summary>
        /// Enum IsGreaterOrEqual for value: IsGreaterOrEqual
        /// </summary>
        [EnumMember(Value = "IsGreaterOrEqual")]
        IsGreaterOrEqual = 4,
        /// <summary>
        /// Enum IsLess for value: IsLess
        /// </summary>
        [EnumMember(Value = "IsLess")]
        IsLess = 5,
        /// <summary>
        /// Enum IsLessOrEqual for value: IsLessOrEqual
        /// </summary>
        [EnumMember(Value = "IsLessOrEqual")]
        IsLessOrEqual = 6    }
}
